utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt

```diff
@@ -231,3 +231,74 @@ annotation class EnableInteractionConfigChangeStateRetention
  * Default value for feature flag corresponding to [EnableInteractionConfigChangeStateRetention].
  */
 const val ENABLE_INTERACTION_CONFIG_CHANGE_STATE_RETENTION_DEFAULT_VALUE = false
+
+/**
+ * Qualifier for the [EnableAppAndOsDeprecation] feature flag that controls whether to enable
+ * app and OS deprecation or not.
+ */
+@Qualifier
+annotation class EnableAppAndOsDeprecation
+
+/**
+ * Default value for the feature flag corresponding to [EnableAppAndOsDeprecation].
+ */
+const val ENABLE_APP_AND_OS_DEPRECATION_DEFAULT_VALUE = false
+
+/**
+ * Qualifier for the platform parameter that contains the version code of the latest available
+ * optional app update, which is used to notify the app that a soft update is available.
+ */
+@Qualifier
+annotation class OptionalAppUpdateVersionCode
+
+/**
+ * Name of the platform parameter that contains the integer version code of the latest available
+ * optional app update.
+ */
+const val OPTIONAL_APP_UPDATE_VERSION_CODE = "optional_app_update_version_code"
+
+/**
+ * Default value for the platform parameter that contains the integer version code of the latest
+ * available optional app update.
+ */
+const val OPTIONAL_APP_UPDATE_VERSION_CODE_DEFAULT_VALUE = 1
+
+/**
+ * Qualifier for the platform parameter that contains the version code of the latest available
+ * forced app update, which is used to notify the app that a mandatory update is available.
+ */
+@Qualifier
+annotation class ForcedAppUpdateVersionCode
+
+/**
+ * Name of the platform parameter that contains the integer version code of the latest available
+ * forced app update.
+ */
+const val FORCED_APP_UPDATE_VERSION_CODE = "forced_app_update_version_code"
+
+/**
+ * Default value for the platform parameter that contains the integer version code of the latest
+ * available forced app update.
+ */
+const val FORCED_APP_UPDATE_VERSION_CODE_DEFAULT_VALUE = 0
```

Comment:
For this & the above, won't these defaults always falsely trigger the deprecation notices? I'd imagine that we'd need the defaults to be the app's local version code by default.

An alternative might be to avoid the default constant and instead pull the version code from the context in the module and use that as the defaults, instead. WDYT?

Reply:
The defaults are always lower or equal to the current version code (which is currently at 1) so I don't think that false deprecation notices would be something we have to worry about. However, avoiding the default constants and using the current version code is also a viable approach.

Reply:
I have updated the code to now use the current app version as the default value for the forced and optional update defaults. I have also added tests for this.